### DISCUSSION TITLE
Error in was-nd-install

### DISCUSSION THREAD

**Discussion by wannesrams** [HCL-TECH-SOFTWARE/connections-automation#8] [Q&A] [answered]
["ERROR: The com.ibm.websphere.ND.v85 package specified in the /opt/IBM/Binaries/logs/was_nd.rsp file cannot be found."], "stdout": "", "stdout_lines": []

All 3 install files are there, unpacked and the rsp file is there as well, the install log gives me the same message but nothing more

**Comment by dejo1307** [comment]
Hey Wannes,

Can you unpack your first installation zip manually, check the repository.config file inside, and see what is offering ID in there?

What happens actually is that: 
- We are using response files for WAS ND (this is the template: https://github.com/HCL-TECH-SOFTWARE/connections-automation/blob/main/roles/third_party/ibm/wasnd/was-nd-install/templates/was_nd.rsp.j2)
- Expected offering ID is here: https://github.com/HCL-TECH-SOFTWARE/connections-automation/blob/54e88d4a969d04c4124cf5538219bba4e06855e7/roles/third_party/ibm/wasnd/was-nd-install/templates/was_nd.rsp.j2#L15 (it is  com.ibm.websphere.ND.v85)

The error you are getting is from installer itself. Can you check that out? If you have in any case different offering ID there, to simply overcome this issue and test it, you can eventually edit that template file from above and change it there (put your own in there). 

I was not aware that there could be different offering IDs for WAS ND 8.5.5 (my logic was: install base, then apply upgrade) but this can be parameterized as well if needed.

I just opened https://github.com/HCL-TECH-SOFTWARE/connections-automation/issues/9 to fix it (I am currently out of office until 4th of January, but will be fixed for sure).

**Reply by wannesrams** [reply]
Odd the repo file only contains this

LayoutPolicy=Composite
LayoutPolicyVersion=0.0.0.1

repository.url.disk1=./disk1

You did mean the WAS_ND_V8.5.5_1_OF_3.zip right?

**Comment by dejo1307** [comment]
Yes... let me just try to figure out if I am talking about wrong file. Can you just grep for word 'offering' in that folder? I am looking where to download it again on my machine since I don't have access here to where I need it :)

**Comment by wannesrams** [comment, marked as answer]
There is actually a repository.xml file in one of the subfolders that seems to match in /md
<?xml version='1.0' encoding='UTF-8'?>
<?repositoryDigest version='0.0.1'?>
<repositoryDigest>
  <repository>
    <property name='LayoutPolicy' value='P1'/>
    <property name='LayoutPolicyVersion' value='0.0.0.1'/>
    <property name='RequireContext' value='EnclosingVolumeRepository:path=..'/>
    <offering id='com.ibm.websphere.BASE.v85' version='8.5.5000.20130514_1044'>
      <information name='%Offering.name' version='8.5.5.0'>
        %Offering.description
      </information>

Your hunch was right, I have the wrong package, it's not the ND package I think. I will fix that and try again but I guess that should work

**Reply by wannesrams** [reply]
Enjoy your time off

**Reply by dejo1307** [reply]
I'm not really WAS expert, I was basing my guesses (I guess that is also why it is currently hardcoded) on the IBM's list of offering IDs for WAS ND but yeah could be it is changing as well. Will definitely recheck it again, but once you get the ID it should work. Update works the same way (default is FP18 if you didn't overwrite it).
